tests/test_models.py

```diff
@@ -814,11 +816,26 @@ def test_model_validate_without_save():
 
 
 def test_modelcontainer_init():
-    img = utils.mk_level1_science_raw()
+    img = utils.mk_level2_image()
     m = datamodels.ImageModel(img)
     mc1 = datamodels.ModelContainer([m])
 
     # initialize with an instance of ModelContainer
     mc2 = datamodels.ModelContainer(mc1)
 
     assert len(mc1) == len(mc2)
+
+
+@pytest.mark.filterwarnings("ignore:ERFA function.*")
+@pytest.mark.parametrize("node", datamodels.model_registry.keys())
+@pytest.mark.parametrize("correct, model", datamodels.model_registry.items())
+def test_model_only_init_with_correct_node(node, correct, model):
+    """
+    Datamodels should only be initializable with the correct node in the model_registry.
+
+    This checks that it can be initiallized with the correct node, and that it cannot be
+    with any other node.
+    """
+    img = create_node(node)
```

Comment:
This test is rather slow because the `testing.factories` constructors can be extremely slow due to lots of random number generation. #193 can make this much more efficient because it provides a `maker_util` that can determine which node to make, that is only available to `testing.factories` at this time.